OpenEdXMobile/src/main/java/org/edx/mobile/services/CourseManager.java

```diff
@@ -134,4 +139,26 @@ public boolean apply(CourseComponent courseComponent) {
             }
         });
     }
+
+    /**
+     * Obtain the enrolled courses data from app level cache and then find the specified course.
+     *
+     * @param courseId    Id of the course.
+     * @return Searched out the course. In case course is not present in
+     * app level cache it will return false.
+     */
+    public Boolean getEnrolledCourseByIdFromAppLevelCache(@NonNull final String courseId) {
+        try {
+            List<EnrolledCoursesResponse> enrolledCoursesResponse =
+                    executeStrict(courseApi.getEnrolledCoursesFromCache());
+            for (EnrolledCoursesResponse course : enrolledCoursesResponse) {
```

Comment:
We are duplicating the functionality of this function. It should be removed and the following should be used.
https://github.com/edx/edx-app-android/blob/c897ba317039e0ad0d77d206ea09b47d44404461/OpenEdXMobile/src/main/java/org/edx/mobile/course/CourseAPI.java#L118-L131